### PR TITLE
Makes pill packets great again!

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Pill packets are now using overlays instead of static sprites. 
As you remove pills from the packet, the sprite will update in real time.
An empty packet will become a trash item, just like a MRE.
Recolors Paracetamol, Ryetalyn, Leporazine and Russian red packets so they are no longer tricordrazine color.
Buffs Paracetamol, Ryetalyn, Leporazine and red russian from 4 >> 8 pills to comply with new sprites.
Pill packets no longer "Containts" in the description.

In order: Bicaridine, Kelotane, Tramadol, Tricordazine, Dylovene, Leporazine, Russian Red, Ryetalyn, Paracematol
image
image

Empty pill packets crumple up just like an MRE would:
image

## Why It's Good For The Game
Pill packets can use some love, this makes it much more readable and reduces annoyance of picking up a pill packet only to see the number 0 on it, realizing it's empty.
Also reduces the amount of half-empty/empty packets spread all over since empty packets become a trash item.
## Changelog
:cl:
add: Crumpled pill packets are biodegradable
balance: Paracetamol, Leporazine, Red Russian, Ryetalyn Packets. 4 Pills >> 8 Pills.
fix: Pill sound now properly plays when you take a pill out of a bottle/packet
imageadd: New pill packet sprites.
/:cl:

### CODE REVIEW

**Review by Kuro020 — APPROVED** [review]

File: code/game/objects/items/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -281,6 +281,15 @@
 	list_reagents = list(/datum/reagent/consumable/honey = 10, /datum/reagent/consumable/nutriment = 0.5, /datum/reagent/consumable/sugar = 2)
 	bitesize = 2
 
+/obj/item/reagent_containers/food/snacks/pillpacket
+	name = "\improper crumpled pill packet"
```
Comment (Lumipharon) [inline code/game/objects/items/reagent_containers/food/snacks.dm:285]:
```suggestion
	name = "crumpled pill packet"
```

File: code/game/objects/items/storage/firstaid.dm [hunk 2]
```diff
@@ -303,7 +303,19 @@
...
+	if(contents.len)
+		var/obj/item/I = contents[1]
+		if(!remove_from_storage(I,user,user))
+			return
+		if(user.put_in_hands(I))
+			to_chat(user, span_notice("You take a pill out of \the [src]."))
+			playsound(user, 'sound/items/pills.ogg', 15, 1)
+	else
+		user.balloon_alert(user, "Empty")
 		return
```
Comment (Lumipharon) [inline code/game/objects/items/storage/firstaid.dm:319]: This is just completely bypassing the parent proc?

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -281,6 +281,15 @@
...
 	bitesize = 2
 
+/obj/item/reagent_containers/food/snacks/pillpacket
+	name = "crumpled pill packet"
+	desc = "Fully biodegradable, some say it's delicious."
+	icon = 'icons/obj/items/trash.dmi'
+	icon_state = "pillpacketempty"
+	list_reagents = list(/datum/reagent/consumable/nutriment = 1) //Done just so that it makes the munching sound
+	tastes = list("syrupy medicine" = 1)
+	bitesize = 0.1
```
Comment (Lumipharon) [inline code/game/objects/items/reagent_containers/food/snacks.dm:291]: Can we not have ungas eating plastic wrappers please

File: code/game/objects/items/storage/pill_packets.dm [hunk 4]
```diff
@@ -2,71 +2,86 @@
...
-	icon_state = "pill_packet"
+	desc = "Contains pills. Once you take them out they don't go back in."
+	icon_state = "packet_canister"
 	cant_hold = list(/obj/item/reagent_containers/pill) //Nada. Once you take the pills out. They don't come back in.
-	storage_slots = 4
+	storage_slots = 8
 	max_w_class = 0
-	max_storage_space = 4
+	max_storage_space = 8
+	greyscale_config = null //So that we get packets and not pill bottles.
```
Comment (Lumipharon) [inline code/game/objects/items/storage/pill_packets.dm:11]: this is already null, unneeded var

File: code/game/objects/items/storage/pill_packets.dm [hunk 5]
```diff
@@ -2,71 +2,86 @@
...
 
+/obj/item/storage/pill_bottle/packet/remove_from_storage(obj/item/item, atom/new_location, mob/user)
+	. = ..()
+	if(!.)
+		return
+	update_icon()
+	if(!contents.len && !gc_destroyed)
+		var/turf/T = get_turf(src)
+		new trash_item(T)
+		qdel(src)
```
Comment (Lumipharon) [inline code/game/objects/items/storage/pill_packets.dm:27]: we don't need to update icon if we're gonna qdel it anyway. So lets rewrite this with an early return with update icon instead.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/pill_packets.dm (hunk 4 above)
Comment (Xander3359) [reply code/game/objects/items/storage/pill_packets.dm:11]: It's not already null

Comment (Xander3359) [reply code/game/objects/items/storage/pill_packets.dm:11]: greyscale_config = /datum/greyscale_config/pillbottle
From the parent /obj/item/storage/pill_bottle

Comment (Xander3359) [reply code/game/objects/items/storage/pill_packets.dm:11]: If I don't null it, the packets turn into bottles

File: code/game/objects/items/reagent_containers/food/snacks.dm (hunk 3 above)
Comment (Xander3359) [reply code/game/objects/items/reagent_containers/food/snacks.dm:291]: Fine

**Review by Lumipharon — APPROVED** [review]

**Comment by CesarVerissimo** [comment]
These sprites are dogshit, please revert.

**Pull request merged**